### DISCUSSION TITLE
Direct unicode input of the most common subscripts and superscripts

### DISCUSSION THREAD

**Discussion by universemaster** [KaTeX/KaTeX#3631] [Ideas]
KaTeX [supports](https://katex.org/docs/supported.html) direct unicode input of many characters which makes sources files much more readable.

For example, this means our source code is not littered with `\lambda` but instead our source simply contains `λ`.

Unicode has a [superscripts and superscripts block](https://en.wikipedia.org/wiki/Unicode_subscripts_and_superscripts#Superscripts_and_subscripts_block) with support for the most commonly used superscripts and subscripts.

However, KaTeX does not treat these characters in any special way.  For example, entering `x² ` is not equivalent to `x^2 ` or `x^{2}`. And similarly  `a₀` is not equivalent to `a_0 ` or `a_{0}`.

KaTeX has a strict mode - which can disable 'features that make writing LaTeX convenient but are not actually supported by (Xe)LaTeX'.

Is there a reason KaTeX cannot or should not interpret say, `x² ` as equivalent to `x^2 ` and `x^{2} ` in non-strict mode - and similarly for all subscript and superscripts available in unicode?

**Comment by universemaster** [comment]
The unicode math package for latex does seem to implement this idea (never used it myself).

See the comments in the source at [unicode math](https://github.com/wspr/unicode-math/blob/master/um-code-sscript.dtx) for some implementation details.

**Reply by edemaine** [reply]
The relevant issue is #1281. But I don't think we knew unicode-math does this. Now that we do, we (a PR) can try to mimic its behavior.
